packages/blade/src/components/Input/PhoneNumberInput/PhoneNumberInput.web.tsx

```diff
@@ -0,0 +1,227 @@
+      dialCode: getDialCodeByCountryCode(selectedCountry),
+      countryCode: selectedCountry,
+    });
+  };
+
+  const onItemClick = ({ name }: { name: string }): void => {
+    const selectedCountry = name as CountryCodeType;
+    setSelectedCountry(selectedCountry);
+    handleOnChange({
+      selectedCountry,
+      name: inputRef.current?.name,
+      value: inputRef.current?.value,
+    });
+    inputRef.current?.focus();
+  };
+
+  return (
+    <BaseInput
+      setInputWrapperRef={(node) => {
+        inputWrapperRef.current = node as HTMLInputElement;
+      }}
+      ref={mergedRef}
+      id="phone-number-input"
+      componentName={MetaConstants.PhoneNumberInput}
+      label={label as string}
+      hideLabelText={!Boolean(label)}
+      labelPosition={labelPosition}
+      defaultValue={defaultValue}
+      value={value}
+      name={name}
+      placeholder={formatPhoneNumber('1234567890', selectedCountry)}
+      trailingIcon={trailingIcon}
+      leadingIcon={leadingIcon}
+      prefix={showDialCode ? getDialCodeByCountryCode(selectedCountry) : undefined}
+      trailingInteractionElement={renderTrailingInteractionElement()}
+      leadingInteractionElement={
+        showCountrySelector ? (
+          <CountrySelector
+            size={size}
+            countryData={countryData}
+            flags={flags}
+            inputWrapperRef={inputWrapperRef}
+            isDisabled={isDisabled}
+            selectedCountry={selectedCountry}
+            onItemClick={onItemClick}
+          />
+        ) : null
+      }
+      onChange={({ name, value }) => {
+        if (value?.length) {
+          // show the clear button when the user starts typing in
+          setShouldShowClearButton(true);
+        }
+        if (shouldShowClearButton && !value?.length) {
+          // hide the clear button when the input field is empty
+          setShouldShowClearButton(false);
+        }
+        handleOnChange({ name, value, selectedCountry });
+      }}
+      onClick={onClick}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      isDisabled={isDisabled}
+      accessibilityLabel={accessibilityLabel}
+      necessityIndicator={necessityIndicator}
+      isRequired={isRequired}
+      validationState={validationState}
+      errorText={errorText}
+      helpText={helpText}
+      successText={successText}
+      // eslint-disable-next-line jsx-a11y/no-autofocus
+      autoFocus={autoFocus}
+      testID={testID}
+      size={size}
+      {...getKeyboardAndAutocompleteProps({
+        type: 'number',
+        keyboardReturnKeyType,
+        autoCompleteSuggestionType,
+        autoCapitalize,
+      })}
+      type="telephone"
+      {...styledProps}
+    />
+  );
+};
+
+/**
+ * PhoneNumberInput is a component that allows users to input phone numbers.
+ * It provides a country selector dropdown to select the country code.
+ *
+ * @example
+ *
+ * ```ts
+ * <PhoneNumberInput />
```

Comment:
should we add basic validation example here? since pretty much every usage would require it along with validation?

Reply:
Validation usecase is already added in storybook example.